### DISCUSSION TITLE
Is Transfer learning possible with NEP?

### DISCUSSION THREAD

**Discussion by Sonti974948** [brucefan1983/GPUMD#624] [Q&A] [answered]
Hi GPUMD team, 

I have trained a machine learning potential using NEP to study the dynamics of gold (Au) in zeolite systems. I would like to conduct the same study for platinum (Pt) in zeolite systems too. 

The current NEP model I trained is purely on Au@zeolite data points (~60,000). Can we add a minimal amount of Pt@zeolite DFT data to the existing MLP (say around 1000 points) to observe good accuracy for even Pt@zeolite systems too? It is too much work to retrain a completely new potential just for Pt@zeolite systems. 

Thank you very much for the help!

**Comment by brucefan1983** [comment, marked as answer]
Currently we do not have transfer learning features. A dataset of 60 000 structures sounds to be very large (I can say it is huge for NEP). 

Do you want to have  (Au+Pt)@zeolite or just Pt@zeolite?

In any case, I would suggest you first simplify the existing training dataset (to about 3000 structures) using, e.g. the farthest-point sampling in calorine (https://calorine.materialsmodeling.org/index.html). Then I think you can substitude Au atoms by Pt atom to create 3000 Pt@zeolite structures and do single-point DFT calculations. This would be a very good starting point for further refining. 

Zheyong

**Reply by Sonti974948** [reply]
This helps a lot. I will try to sample a smaller subset for Pt@zeolite and try to retrain the model. Thank you!

**Comment by brucefan1983** [comment]
Actually we might have transfer learning if you only need Pt@zeolite.

As you said, you would like to create about 1000 Pt@zeolite structures. Perhaps you can try to restart your traning from your Au@zeolite model? You only need to take the previous nep.in and nep.restart and change the species Au to Pt in nep.in.

**Reply by Sonti974948** [reply]
Right now I only want to extend this to Pt@zeolite as a test. But, if successful, I would actually want to extend the same MLP to all the other precious metals in zeolites.

**Reply by brucefan1983** [reply]
OK, I think you really need to control the size of the dataset. In my experiences, NEP does not need a huge amount of training data, but if you want to have a balanced unified model, you need to have balanced data for each subset.
